docs/arcaflow/workflows/flow-control.md

```diff
@@ -2,6 +2,258 @@
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !expr $.steps.my_step.outputs.success
+```
+
+But oh no! The workflow fails with the error `all outputs marked as unresolvable` when the step is disabled.
+Why is this? It's simple. The step did not run, as instructed, and therefore the output `workflow_success` cannot get the information it needs. See the next section for methods of handling this situation.
+
+#### Graceful handling of disabled steps
+
+When steps are disabled, they no longer emit their step output. To work around this, the workflow output (or a following step) must have an OR dependency on either the step's disable output or on another step that has opposite enable logic.
+
+Here is a visual demonstrating two steps that have opposite logic:
+```mermaid
+stateDiagram-v2
+  classDef tag font-style:italic;
+  classDef step font-weight:bold;
+
+  state if_state <<choice>>
+  Step1:::step: Step 1
+  [*] --> Step1
+  Step1 --> if_state
+  Step2:::step: Step 2
+  Step3:::step: Step 3
+  if_state --> Step2: !expr $.step1.output_1 == true
+  if_state --> Step3: !expr $.step1.output_1 == false
+  oneof:::tag: !oneof
+  Step2 --> oneof
+  Step3 --> oneof
+  oneof --> [*]
+```
+
+Here is a visual demonstrating utilizing the disabled output:
+```mermaid
+stateDiagram-v2
+  classDef tag font-style:italic;
+  classDef step font-weight:bold;
+
+  input
+  [*] --> Step1:::step
+  input --> Step1: !expr $.input.enabled
+  or_disabled:::tag: !ordisabled
+  Step1 --> or_disabled: success
+  Step1 --> or_disabled: Disabled
+  or_disabled --> [*]
+```
+
+The way this works is that either of the two paths can create a valid output, allowing the workflow to continue past the disabled steps.
+
+##### How to do this in a workflow
+
+There are two tags that create the described OR dependency.
+
+| Tag           | Description                                      |
+|---------------|--------------------------------------------------|
+| `!oneof`      | A general use tag to depend on any of the inputs |
+| `!ordisabled` | A special case of `!oneof` that automatically handles the disabled case |
+
+##### How to use `!oneof`
+
+To use `!oneof` for graceful handling of disabled steps, the oneof should depend on either ONE OF the success output or the disabled output.
+
+The oneof tag is a method of creating a schema `one_of_string` type from values present in the workflow.
+
+The syntax of `!oneof` is:
+- Following the tag `!oneof`, create a new YAML section (map) by moving to a new line. That section should contain two properties:
```

Comment:
"Moving to" sounds weirdly special-case when you're creating a workflow file. Maybe "starting a new line"?

Reply:
I changed it to "starting an indented new line".